pkg/aws/cloud.go

```diff
@@ -126,6 +130,37 @@ func NewCloud(cfg CloudConfig, metricsRegisterer prometheus.Registerer) (Cloud,
 	}, nil
 }
 
+func getVPC(metadata services.EC2Metadata, ec2Service services.EC2) string {
```

Comment:
this breaks the existing experience as it hide the error if metadata is not available since it hides the error.
I think change to below will be better:
```
inferVPCID(metadata, ec2Service) (vpcID, error) {
	var errors []err
	vpcID, err := metadata.vpcID()
	if err == nil {
		return vpcID
	}
	errors = append(errors, errors.Wrap(err, "cannot infer VPCID from metadata"))
	vpcID, err := inferVPCIDFromNodeName(...)
	if err == nil {
		return vpcID
	}
	errors = append(errors, err)
	return "", errors.NewAggregate(errors) # https://github.com/kubernetes/apimachinery/blob/478dd6ed1ec9b4cc3459dcd4f1d772013ac0103c/pkg/util/errors/errors.go#L46
}

```

Reply:
Ouh. Apimachinery has it's own error aggregate? Today I learnt. Will fix.